src/ZSPI.cpp

```diff
@@ -338,9 +344,12 @@ int ZSPI::startTransfer(const uint8_t *txBuffer, uint32_t txSize, uint8_t *rxBuf
     }
     else
     {
-        return 0; // nothing to do
+        res = HAL_OK;
     }
 
+    if (doneHandler)
```

Comment:
I guess you do this check just to minimise interference with interrupts?

Reply:
yeah, if there is no handler, the completed even will go through the event bus and we'll be golden